plugins/lookup/binary_file.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2025, Felix Fontein <felix@fontein.de>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+DOCUMENTATION = r"""
+name: binary_file
+author: Felix Fontein (@felixfontein)
+short_description: Read binary file and return it Base64 encoded
+version_added: 11.2.0
+description:
+  - This lookup returns the contents from a file on the Ansible controller's file system.
+  - The file is read as a binary file and its contents are returned Base64 encoded.
+    This is similar to using P(ansible.builtin.file#lookup) combined with P(ansible.builtin.b64encode#filter),
+    except that P(ansible.builtin.file#lookup) does not support binary files as it interprets the contents as UTF-8,
+    which can cause the wrong content being Base64 encoded. It also will print a deprecation message since ansible-core 2.19.
+options:
+  _terms:
+    description:
+      - Paths of the files to read.
+      - Relative paths will be searched for in different places. See R(Ansible task paths, playbook_task_paths) for more details.
+    required: true
+    type: list
+    elements: str
+  not_exist:
+    description:
+      - Determine how to react if the specified file cannot be found.
+    type: str
+    choices:
+      error: Raise an error.
+      empty: Return an empty string for the file.
+      empty_str:
+        - Return the string C(empty) for the file.
+        - This cannot be confused with Base64 encoding due to the missing padding.
+    default: error
+notes:
+  - This lookup does not understand 'globbing' - use the P(ansible.builtin.fileglob#lookup) lookup instead.
+seealso:
+  - plugin: ansible.builtin.b64decode
+    plugin_type: filter
+    description: >-
+      The b64decode filter can be used to decode Base64 encoded data.
+      Note that Ansible cannot handle binary data, the data will be interpreted as UTF-8 text!
+  - plugin: ansible.builtin.file
```

Comment:
Maybe mention the deprecation here, as a warning, something like "Beware of the deprecation in 2.19. There be dragons."